src/AspNet.Hosting.Extensions/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,263 @@
+            [NotNull] PathString pathMatch,
+            [NotNull] Action<IApplicationBuilder> configuration,
+            [NotNull] Action<IServiceCollection> serviceConfiguration)
+        {
+            return app.IsolatedMap(
+                pathMatch,
+                configuration,
+                services => {
+                    serviceConfiguration(services);
+                    return services.BuildServiceProvider();
+                });
+        }
+
+        /// <summary>
+        ///   If the request path starts with the given <see cref="pathMatch"/>, execute the app configured via
+        ///   <see cref="configuration"/> parameter instead of continuing to the next component in the pipeline.
+        ///   The new app will get an own newly created <see cref="ServiceCollection"/> and will not share the
+        ///   <see cref="ServiceCollection"/> from the originating app.
+        /// </summary>
+        /// <param name="app">The application builder to register the isolated map with.</param>
+        /// <param name="pathMatch">The path to match. Must not end with a '/'.</param>
+        /// <param name="configuration">The branch to take for positive path matches.</param>
+        /// <param name="serviceConfiguration">A method to configure the newly created service collection.</param>
+        /// <returns>The new pipeline with the isolated middleware configured.</returns>
+        public static IApplicationBuilder IsolatedMap(
+            [NotNull] this IApplicationBuilder app,
+            [NotNull] PathString pathMatch,
+            [NotNull] Action<IApplicationBuilder> configuration,
+            [NotNull] Func<IServiceCollection, IServiceProvider> serviceConfiguration) {
+            if (pathMatch.HasValue && pathMatch.Value.EndsWith("/", StringComparison.Ordinal))
+            {
+                throw new ArgumentException("The path must not end with a '/'", nameof(pathMatch));
+            }
+
+            return app.Isolated(
+                subApp => subApp.Map(pathMatch, configuration),
+                serviceConfiguration);
+        }
+        /// <summary>
+        ///   Creates a new isolated application builder which gets its own <see cref="ServiceCollection"/>, which only
+        ///   has the default services registered. It will not share the <see cref="ServiceCollection"/> from the
+        ///   originating app. The isolated map will be configured using the configuration methods of the
+        ///   <see cref="TStartup"/> class.
+        /// </summary>
+        /// <typeparam name="TStartup">
+        ///   The startup class used to configure the new app and the service collection.
+        /// </typeparam>
+        /// <param name="app">The application builder to create the isolated app from.</param>
+        /// <returns>The new pipeline with the isolated application integrated.</returns>
+        public static IApplicationBuilder Isolated<TStartup>(
```

Comment:
Have you considered naming it `Isolate`? Looks like the other extensions (Use, Map, Run) are all verbs.